src/pgen/cluster.cpp

```diff
@@ -405,4 +432,101 @@ void ProblemGenerator(MeshBlock *pmb, parthenon::ParameterInput *pin) {
   } // END if(hydro_pkg->Param<Fluid>("fluid") == Fluid::glmmhd)
 }
 
+void UserWorkBeforeOutput(MeshBlock *pmb, ParameterInput *pin) {
+  // get hydro
+  auto pkg = pmb->packages.Get("Hydro");
+  const Real gam = pin->GetReal("hydro", "gamma");
+  const Real gm1 = (gam - 1.0);
+
+  // get prim vars
+  auto &data = pmb->meshblock_data.Get();
+  auto const &prim = data->Get("prim").data;
```

Comment:
"""
Fairly sure. Because this function is called for the output (i.e., outside of a cycle) and the design/assumption is that data is consistent at the beginning/end of a cycle.
"""
I'm assuming that includes ghost zones too so this isn't the cause for Ascent meshblock boundary artifacts.

"""
It should be fairly straightforward to automatically create a flowchart with a small PR in Parthenon that dumps the task list in a format that is understood by some graph library.
Any volunteers? ;)
"""
Sounds interesting to me at least! I might take a stab at it

Reply:
>> """ Fairly sure. Because this function is called for the output (i.e., outside of a cycle) and the design/assumption is that data is consistent at the beginning/end of a cycle. """

> I'm assuming that includes ghost zones too so this isn't the cause for Ascent meshblock boundary artifacts.

Correct, ghost cells are up-to-date when outputs are called.

>> """ It should be fairly straightforward to automatically create a flowchart with a small PR in Parthenon that dumps the task list in a format that is understood by some graph library. Any volunteers? ;) """

> Sounds interesting to me at least! I might take a stab at it

For reference, my first attempt/thoughts https://github.com/parthenon-hpc-lab/parthenon/issues/880